src/serac/physics/heat_transfer.hpp

```diff
@@ -285,6 +285,64 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
     cycle_ += 1;
   }
 
+  /**
+   * @brief Functor representing the integrand of a thermal material.  Material type must be
+   * a functor as well.
+   */
+  template <typename MaterialType>
+  struct ThermalMaterialIntegrand {
+    /**
+     * @brief Construct a ThermalMaterialIntegrand functor with material model of type `MaterialType`.
+     * @param[in] material A functor representing the material model.  Should be a functor, or a class/struct with
+     * public operator() method.  Must NOT be a generic lambda, or serac will not compile due to static asserts below.
+     */
+    ThermalMaterialIntegrand(MaterialType material) : material_(material) {}
+
+    // Due to nvcc's lack of support for generic lambdas (i.e. functions of the form
```

Comment:
👍

Reply:
this comment is misleading, as `nvcc` does support generic lambdas, and it also supports their use in cuda kernels. For example,

https://godbolt.org/z/TM8zacGcM

What it doesn't support is the corner case of generic, extended lambdas.

Reply:
I agree, this should be specified for extended lambdas.